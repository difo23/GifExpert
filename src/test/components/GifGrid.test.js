import React from 'react';
import Gifgrid from '../../components/GifExpertApp/GifGrid';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test GifGrid Component', () => {
	const CATEGORY = 'Bob Sponge';

	test('should default data and loading true', () => {
		let gifs = {
			data: [],
			loading: true
		};

		useFetchGifs.mockReturnValue(gifs);

		const wrapper = shallow(<Gifgrid category={CATEGORY} />);

		expect(wrapper.find('p').exists()).toBe(true);
		expect(wrapper.find('GifItem').length).toBe(gifs.data.length);
	});

	test('should get data and loading false', () => {
		let gifs = {
			data: [
				{
					id: '1234',
					url: 'https://localhost:3000/algo.png',
					title: 'Pruebas unitarias'
				},
				{
					id: '12345',
					url: 'https://localhost:3000/algo.png',
					title: 'Pruebas unitarias'
				}
			],
			loading: false
		};

		useFetchGifs.mockReturnValue(gifs);

		const wrapper = shallow(<Gifgrid category={CATEGORY} />);
		let progress = wrapper.find('p');

		expect(wrapper).toMatchSnapshot();
		expect(progress.exists()).toBe(false);
		expect(wrapper.find('GifItem').length).toBe(gifs.data.length);
	});
});
