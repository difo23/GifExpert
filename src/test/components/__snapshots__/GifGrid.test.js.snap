// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGrid Component should get data and loading false 1`] = `
<div>
  <h3>
    Bob Sponge
  </h3>
  <ul
    className="card-grid"
  >
    <GifItem
      key="1234"
      title="Pruebas unitarias"
      url="https://localhost:3000/algo.png"
    />
    <GifItem
      key="12345"
      title="Pruebas unitarias"
      url="https://localhost:3000/algo.png"
    />
  </ul>
</div>
`;
